llama_index/storage/index_store/utils.py

```diff
@@ -15,4 +15,4 @@ def json_to_index_struct(struct_dict: dict) -> IndexStruct:
     type = struct_dict[TYPE_KEY]
     data_dict = struct_dict[DATA_KEY]
     cls = INDEX_STRUCT_TYPE_TO_INDEX_STRUCT_CLASS[type]
-    return cls.from_dict(data_dict)
+    return cls.from_json(data_dict)
```

Comment:
Could we try/catch `TypeError`, and fallback to loading the data as dict (to remain compatible with users who saved index in the past).

Reply:
And add a note saying data blobs were stored as dict, but now is serialized to string first.